### ISSUE TITLE
Error: `collectionType` must inherit from Backbone.Collection

### ISSUE THREAD

**Issue by azproduction** [dhruvaray/spa-eye#13]
1. Enable Backbone Eye
2. Open http://m.taxi.yandex.ru/
3. Error 

Backbone+Backbone.Relational

**Comment by dhruvaray** [comment]
Any chance, we could get access to a staging URL (with unminfied js)? I can share my email with you.

**Comment by azproduction** [comment]
Your code monkey-patchs `Backbone.Collection` that is why backbone-relational throws this error

``` js
if ( this.collectionType !== Backbone.Collection && !( this.collectionType.prototype instanceof Backbone.Collection ) ) {
    throw new Error( '`collectionType` must inherit from Backbone.Collection' );
}
```

 https://github.com/PaulUithol/Backbone-relational/blob/master/backbone-relational.js#L861-L863

``` js
Backbone.Collection + '';
function () {
    var event = new CustomEvent('Backbone_Eye:ADD', {'detail':{data:this, entity_type:entity}});
    ...
}
```

Maybe we can find elegant way of patching constructors.

**Comment by azproduction** [comment]
btw, can Backbone Eye handle isolated Backbone (in closure)?

**Comment by dhruvaray** [comment]
> > btw, can Backbone Eye handle isolated Backbone (in closure)?

Yes. You need to switch on deep-check option - http://dhruvaray.github.io/spa-eye/tips.html

**Comment by dhruvaray** [comment]
> > Your code monkey-patchs Backbone.Collection that is why backbone-relational throws this error

Yes. But that by itself should not be a problem.

BB-Relational seems to be working per-se. If you run [click on orange play icon] the first example (as well as the other examples) on http://backbonerelational.org/ with Backbone-Eye ON, it picks up the collections/models correctly in-spite of passing through the code you mention.

There must be some other customization (specific to your app - perhaps) which is causing it to break and not recognize it to be a `Collection`. It would be easy to figure out if I had access to an unminified version of your site.

**Comment by azproduction** [comment]
> unminified version of your site.

This is impossible :( I can not even give you a source map.

> There must be some other customization

This project uses CommonJS modules and scripts are compiled with LMD (some kind of browserify). Backbone is in that bundle too, but prints itself into globals. Every module accesses backbone via `require('Backbone')`.

**Comment by dhruvaray** [comment]
Do you have any customizations on top of Backbone-Relational? Can you alternatively paste/share the (unminified js  source) +/- 5 lines from the point of error?

**Comment by azproduction** [comment]
Backbone-relational.js 0.8.0 http://pastebin.com/GAnDb8iX

**Comment by dhruvaray** [comment]
BB-relational v0.8.0 also seems to work with the site examples... Can you do a pastebin of a couple of lines where you instantiate your models/collections?

**Comment by azproduction** [comment]
``` js
var Backbone = require('backbone'),
    _ = require('_');

require('relational');

var AbstractModel = Backbone.RelationalModel.extend({
    //...
    sync: function (method, model, options) {
        return Backbone.sync.call(this, 'create', model, options);
    },
    //...
});

module.exports = AbstractModel;
```

``` js
var Backbone = require('backbone'),
    _ = require('_');

var urlError = function () {
    throw new Error('A "urlRoot" property or function must be specified');
};

var AbstractCollection = Backbone.Collection.extend({
    protocolVersion: '3.0',

    url: function () {
        var base = _.result(this, 'urlRoot') || urlError();

        return '/' + this.protocolVersion + base + '/';
    },

    sync: function (method, model, options) {
        return Backbone.sync.call(this, 'create', model, options);
    },

    save: function () {
        throw new Error('Method save() is not allowed. Use fetch() instead');
    }
});

module.exports = AbstractCollection;
```

**Comment by dhruvaray** [comment]
Models which have a 1:M (or 1:1) relations with 
1. objects
2. ask_feedback, info, orders, phones
3. requirements, hotspots
4. route

via bb-rel seems to have come out correctly. Are there any other model definitions (with relations)? Do you do anything special with those ones? Can you paste their initialization/defn code?
